erpnext/quality_management/doctype/quality_procedure/quality_procedure.py

```diff
@@ -16,18 +16,10 @@ def before_save(self):
 				self.is_group = 1
 
 	def on_update(self):
-		for data in self.procedure_step:
-			if data.procedure == 'Procedure' and data.procedure_name != '':
-				doc = frappe.get_doc("Quality Procedure", data.procedure_name)
-				doc.parent_quality_procedure = self.name
-				doc.save()
+		set_parent(self)
```

Comment:
`self` is reserved for class methods. You should put the `set_parent` function inside the `QualityProcedure` class, and call it using `self.set_parent()`.

Reply:
Fixed